command/runner.go

```diff
@@ -84,3 +85,29 @@ func (r *Runner) Command(name string, args *CommandArgs, opts ...pulumi.Resource
 
 	return remote.NewCommand(r.e.Ctx, r.namer.ResourceName("cmd", name), args.toRemoteCommandArgs(r.connection), opts...)
 }
+
+type LocalRunner struct {
```

Comment:
I would like to change this to implement a `Runner` interface, used by both the remote runner and the local runner. This would allow us to use the same code path in this PR for provisioning local development environments as well.

Would this be suitable?

Reply:
What is preventing from using `Runner` ? Would you need to add a `isLocal` or `flavour` property ?

Reply:
You could also use a common interface and implement common functions in `LocalRunner` and `RemoteRunner`

Reply:
From what I understand `Runner` runs commands on the remote instance provisioned by pulumi.

In this case I am trying to run commands on the machine where pulumi itself is running. I also, need these commands to be tracked by pulumi to gracefully handle update scenarios.

The main issue I am running into with a common interface is that the `Command` object for the local runner and the remote runner are both different. I am not sure how I can implement a scenario given this.

Reply:
Implementation looks legit to me. We could introduce an interface to wrap both, but IMHO may not be worth it ATM.